src/riak_kv_pb_timeseries.erl

```diff
@@ -111,37 +111,44 @@ encode(Message) ->
                      {reply, #tsqueryresp{} | #rpberrorresp{}, #state{}}.
 process(#tsputreq{rows = []}, State) ->
     {reply, #tsputresp{}, State};
-process(#tsputreq{table = Table, columns = _Columns, rows = Rows}, State) ->
+process(#tsputreq{table = Table, columns = _Columns, rows = EncodedRows}, State) ->
     Mod = riak_ql_ddl:make_module_name(Table),
-    Data = riak_pb_ts_codec:decode_rows(Rows),
-    %% validate only the first row as we trust the client to send us
-    %% perfectly uniform data wrt types and order
-    case (catch Mod:validate_obj(hd(Data))) of
-        true ->
-            %% however, prevent bad data to crash us
-            try
-                case put_data(Data, Table, Mod) of
-                    0 ->
-                        {reply, #tsputresp{}, State};
-                    ErrorCount ->
-                        EPutMessage = flat_format("Failed to put ~b record(s)", [ErrorCount]),
-                        {reply, make_rpberrresp(?E_PUT, EPutMessage), State}
-                end
-            catch
-                Class:Exception ->
-                    lager:error("error: ~p:~p~n~p", [Class,Exception,erlang:get_stacktrace()]),
-                    Error = make_rpberrresp(?E_IRREG, to_string({Class, Exception})),
-                    {reply, Error, State}
-            end;
-        false ->
-            {reply, make_rpberrresp(?E_IRREG, "Invalid data"), State};
-        {_, {undef, _}} ->
-            BucketProps = riak_core_bucket:get_bucket(table_to_bucket(Table)),
-            {reply, missing_helper_module(Table, BucketProps), State}
+
+    case riak_pb_ts_codec:decode_rows(EncodedRows) of
+        {error, {bad_ts_cell, BadCell=#tscell{}}} ->
```

Comment:
If we do change the last `decode_cell` function head to match any value, not just `tscell` records, this line (and similar) should also be changed to drop the match.